src/ansys/fluent/core/quantity/dimensions.py

```diff
@@ -20,31 +22,32 @@ def __init__(
         self, units: str = None, dimensions: list = None, unit_sys: str = None
     ):
         if units and dimensions:
-            raise ValueError(
-                "Dimensions only accepts 1 of the following: units, dimensions."
-            )
+            raise DimensionsError.EXCESSIVE_PARAMETERS()
 
         self._units_table = q.UnitsTable()
-        unit_sys = unit_sys or "SI"
+        unit_sys = unit_sys or self._units_table.unit_systems["SI"]
 
         if units is not None:
             self._unit = units
             self._dimensions = self._units_to_dim(units=units)
 
         if dimensions:
+            if len(dimensions) > 9:
```

Comment:
Let's expose this dimensions list size (=9) from this dimensions module. Then we can reference it here, in unit_system, and wherever else needed.

Reply:
Added static method to Dimensions to access max dimensions list length (9). exp -> `Dimensions.max_dim_len()`

https://github.com/ansys/pyfluent/blob/16bfd1f51b57625263ffefce7cf960c6691ef305/src/ansys/fluent/core/quantity/dimensions.py#L132-L135